examples/first.cpp

```diff
@@ -1,59 +1,186 @@
 #include "loops/loops.hpp"
+#include <algorithm>
 #include <vector>
 #include <iostream>
 
-typedef int (*justcompiled_t)(const int* ptr, int64_t n,
+typedef int (*minmaxpos_t)(const int* ptr, int64_t n,
                             int* minpos, int* maxpos);
-loops::Func gencode(loops::Context& ctx)
+loops::Func genminmaxloc(loops::Context& CTX)
 {
     using namespace loops;
     IReg ptr, n, minpos_addr, maxpos_addr;
-    ctx.startFunc("foo", {&ptr, &n, &minpos_addr, &maxpos_addr});
-    IReg i = ctx.const_(0);
-    IReg minpos = ctx.const_(0);
-    IReg maxpos = ctx.const_(0);
+    CTX.startFunc("minmaxloc", {&ptr, &n, &minpos_addr, &maxpos_addr});
+    IReg i = CTX.const_(0);
+    IReg minpos = CTX.const_(0);
+    IReg maxpos = CTX.const_(0);
     IReg minval = load_<int>(ptr);
     IReg maxval = minval;
-    IReg elemsize = ctx.const_((int)sizeof(int));
-    n *= elemsize;
-    ctx.do_();
+    n *= sizeof(int);
+    CTX.do_();
         IReg x = load_<int>(ptr, i);
-        ctx.if_(x < minval);
+        CTX.if_(x < minval);
             minval = x;
             minpos = i;
-        ctx.endif_();
-        ctx.if_(x > maxval);
+        CTX.endif_();
+        CTX.if_(x > maxval);
             maxval = x;
             maxpos = i;
-        ctx.endif_();
-        i += elemsize;
-    ctx.while_(i < n);
+        CTX.endif_();
+        i += sizeof(int);
+    CTX.while_(i < n);
+    IReg elemsize = CTX.const_(sizeof(int));
     minpos /= elemsize;
     maxpos /= elemsize;
     store_<int>(minpos_addr, minpos);
     store_<int>(maxpos_addr, maxpos);
-    ctx.return_(ctx.const_(0));
-    ctx.endFunc();
-    return ctx.getFunc("foo");
+    CTX.return_(0);
+    CTX.endFunc();
+    return CTX.getFunc("minmaxloc");
 }
 
+//This implementation(loops and reference) is taken from wikipedia: https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm
+typedef void (*bresenham_t)(uint8_t* canvas, int64_t w, int64_t x0, int64_t y0, int64_t x1, int64_t y1, uint64_t filler);
```

Comment:
please, rename first.cpp into minmaxloc.cpp and move bresenham part into a separate example